sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -16,69 +16,55 @@ using namespace Azure::Core;
 TEST(Context, Basic)
 {
   Context context;
-  auto& valueT = context["key"];
+  auto valueT = context["key"];
   EXPECT_FALSE(context.HasKey(""));
   EXPECT_FALSE(context.HasKey("key"));
-
-  auto kind = valueT.Alternative();
-  EXPECT_TRUE(kind == ContextValue::ContextValueType::Undefined);
+  EXPECT_EQ(valueT, nullptr);
 }
 
 TEST(Context, BasicBool)
 {
   Context context;
   // New context from previous
-  auto c2 = context.WithValue("key", true);
-  auto& valueT = c2["key"];
-  auto value = valueT.Get<bool>();
-  EXPECT_TRUE(value == true);
-
-  auto kind = valueT.Alternative();
-  EXPECT_TRUE(kind == ContextValue::ContextValueType::Bool);
+  bool value(true);
+  auto c2 = context.WithValue("key", reinterpret_cast<void*>(&value));
+  auto valueVoidP = c2["key"];
+  auto valueT = *reinterpret_cast<bool*>(valueVoidP);
```

Comment:
If we set this value in one function and get this value from another function, and the original value is already destructed on stack, this will crash